catalog/async/catalog-async-data/src/main/java/org/codice/ddf/catalog/async/data/impl/LazyProcessResourceImpl.java

```diff
@@ -180,25 +180,20 @@ public boolean isModified() {
 
   @Override
   public synchronized void close() {
-    try {
-      if (streamCloser != null) {
-        streamCloser.close();
-      }
-    } catch (IOException e) {
-    } // suppress exceptions
+    IOUtils.closeQuietly(streamCloser);
```

Comment:
❗️ What happens if `getInputStream()` is never called but a method like `getName()` which initializes the input stream is called? As far as I can tell, the stream would never be closed. 

We might want to set it to `null` in the `getInputStream()` once used before being closed and add some code here to close it if not `null`. Make sure to `null` it first.

Reply:
Yea the input stream wouldn't be closed. We could have the `streamCloser` register the input stream and if it's not closed in `getInputStream()` then it will be closed in `close()`. In the case that someone doesn't call, `getInputStream` I'm not sure how likely they would be to call close. In that case the input stream will probably just end up being garbage collected (though I hope this wouldn't happend). Sorry, but I'm a little confused by the comment, what do you think we should set to null?

Reply:
That would cause a double close in case where the `getInputStream()` is called. Some input streams throw IOExceptions when that happens but since we ignore exceptions anyway, that might not be an issue